llvm/lib/TableGen/Record.cpp

```diff
@@ -1577,6 +1576,8 @@ const Init *BinOpInit::resolveReferences(Resolver &R) const {
     }
   }
 
+  const Init *rhs = RHS->resolveReferences(R);
+
```

Comment:
Nit: As long as we're in the area, can we update the non-conforming `rhs` and `lhs` to, e.g., `NewRHS` and `NewLHS`?

Otherwise, LGTM.

Reply:
Yeah I was thinking about the same thing. It's fixed now.